batchflow/research/research.py

```diff
@@ -203,32 +246,13 @@ def get_metrics(self, pipeline, metrics_var, metrics_name, *args,
             include execution information to log file or not
         """
         name = pipeline + '_' + metrics_var
+        name += str(len([item for item in self.executables if item.startswith(name)]))
         returns = returns or metrics_name
         self.add_callable(get_metrics, *args, name=name, execute=execute, dump=dump, returns=returns,
                           on_root=False, logging=logging, pipeline=RP(pipeline),
                           metrics_var=metrics_var, metrics_name=metrics_name, agg=agg, **kwargs)
         return self
 
-    def init_domain(self, domain=None, n_configs=None, n_reps=1, repeat_each=100):
```

Comment:
Maybe, you can leave this function, and call it in `__init__` if the domain was passed at initialization?